SIL.Scripture/VerseRef.cs

```diff
@@ -302,14 +302,14 @@ static bool TryGetVerseNum(string verseStr, out short vNum)
 			for (int i = 0; i < verseStr.Length; i++)
 			{
 				char ch = verseStr[i];
-				if (!char.IsDigit(ch))
+				if (romanOnly ? (ch < '0' || ch > '9') : !char.IsDigit(ch))
```

Comment:
Looking at the documentation for [`char.IsDigit/char.IsNumber`](https://docs.microsoft.com/en-us/dotnet/api/system.char.isnumber?view=net-5.0#System_Char_IsNumber_System_Char_) it seems our method here won't work for surrogate pairs (e.g. Chakma digits), and it won't work for numbers that don't have the DecimalDigitNumber category (e.g. Aegean - probably not very relevant for us, but I'm sure there are other similar numbering systems). Cf. <http://unicode.org/pending/properties.html>, [UnicodeData](http://www.unicode.org/Public/UCD/latest/ucd/UnicodeData.txt)

I'm not sure we have to fix this now, but it would be good to document it by creating a GitHub issue and possibly (ignored) failing tests cases.
If you don't want to

Reply:
@ermshiperete Writing some tests for this, but I'm having trouble finding an example of Chakma (or any other script) using a surrogate pair numeral to test against. Could you provide one?

Reply:
I'd think "\U0001113A\U00011138" would be one (42; CHAKMA DIGIT FOUR; CHAKMA DIGIT TWO)

Reply:
Thanks. Wrote the tests and created #1000.